addons/mail/models/mail_message_reaction.py

```diff
@@ -22,3 +23,18 @@ def init(self):
     _sql_constraints = [
         ("partner_or_guest_exists", "CHECK((partner_id IS NOT NULL AND guest_id IS NULL) OR (partner_id IS NULL AND guest_id IS NOT NULL))", "A message reaction must be from a partner or from a guest."),
     ]
+
+    def _to_store(self, store: Store):
+        for (message_id, content), reactions_list in groupby(
+            self, lambda r: (r.message_id, r.content)
+        ):
+            reactions = self.env["mail.message.reaction"].union(*reactions_list)
+            data = {
+                "id": f"{message_id.id}, {content}",
+                "content": content,
+                "count": len(reactions),
+                "personas": Store.many_ids(reactions.guest_id)
```

Comment:
Also not related to this PR, but we're not very smart here. We're sending reactions as group rather than individually for performance and scaling reasons, and then we're sending all the persona anyway.
We need to stop doing this for reactions to scale to 1000+ per message, but that requires other changes in code.